### PR TITLE
Fix greedy bestjoin

### PR DESCRIPTION
**Pull request by bhuvnesh2703**
Fix Bug in Greedy Join Algo while picking best join

The goal of `PickBestJoin` is to find the best component out of all the
available component which gives the minimum cardinality with the existing
`m_pcompResult`. However, `m_pcompResult` was getting updated multiple 
times for the join candidate set.
`m_pcompResult` should only be updated once after it has iterated over all 
the candidate component and found the best  join combination.

Also, have enabled Greedy Algorithm in exhaustive mode as well.

### CODE REVIEW

File: libgpopt/src/xforms/CJoinOrderGreedy.cpp [hunk 1]
```diff
@@ -265,36 +265,41 @@ CJoinOrderGreedy::PickBestJoin
...
 		// pick the component which will give the lowest cardinality
 		if (NULL == pcompBestComponent || dRows < dMinRows)
 		{
 			dMinRows = dRows;
 			best_comp_idx = iter.Bit();
 			pcompBestComponent = pcompCurrent;
-			m_pcompResult->Release();
-			m_pcompResult = pcompTemp;
-			m_pcompResult->AddRef();
+			pcompTemp->AddRef();
```
Comment (hardikar) [inline libgpopt/src/xforms/CJoinOrderGreedy.cpp]:
```
pcompTemp->AddRef();
CRefCount::SafeRelease(pcompBest);
```
This should probably go outside of the loop.

File: libgpopt/src/xforms/CXformExpandNAryJoinGreedy.cpp [hunk 2]
```diff
@@ -69,21 +69,6 @@ CXformExpandNAryJoinGreedy::Exfp
 	)
 	const
 {
-	COptimizerConfig *poconf = COptCtxt::PoctxtFromTLS()->GetOptimizerConfig();
```
Comment (hardikar) [inline libgpopt/src/xforms/CXformExpandNAryJoinGreedy.cpp]: Please separate this out into a different commit. This is a feature on its own.

File: libgpopt/src/xforms/CJoinOrderGreedy.cpp (hunk 1 above)
Comment (bhuvnesh2703) [reply libgpopt/src/xforms/CJoinOrderGreedy.cpp]: As discussed this can't be moved outside.

File: libgpopt/src/xforms/CXformExpandNAryJoinGreedy.cpp (hunk 2 above)
Comment (bhuvnesh2703) [reply libgpopt/src/xforms/CXformExpandNAryJoinGreedy.cpp]: Made separate commits

File: libgpopt/src/xforms/CJoinOrderGreedy.cpp (hunk 1 above)
Comment (iyerr3) [reply libgpopt/src/xforms/CJoinOrderGreedy.cpp]: Another option is to compute the best `pCompBestComponent` and then call `pcompCombine` a final time outside the loop to update `m_pcompResult`. This way the objective of the loop is to compute the best component and then we add that into our result. Not sure about the cost of making an extra `pcomCombine` call - if it's an expensive operation then we can stick with our current approach of maintaining `pcompBest`.
